### PR TITLE
[refactor](wg&memtracker) using weak ptr to delete memtracker and query context automatically

### PR DESCRIPTION
**Pull request by yiguolei**
## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by yiguolei** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

File: be/src/runtime/load_channel.cpp [hunk 1]
```diff
@@ -85,12 +84,6 @@ LoadChannel::~LoadChannel() {
         rows_str << ", index id: " << entry.first << ", total_received_rows: " << entry.second.first
                  << ", num_rows_filtered: " << entry.second.second;
     }
-    if (_need_release_memtracker) {
```
Comment (wangbo) [inline be/src/runtime/load_channel.cpp:88]: How count load channel memory now?

Comment (yiguolei) [reply be/src/runtime/load_channel.cpp:88]: This  PR does not affect how to count low channel memory, it only changed how to remove memtracker from workload group.

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**